File: Projects/03-Calculator/C/main.c
#include <stdio.h>
#include <string.h>

int add(int a, int b);
int sub(int a, int b);
int mul(int a, int b);
int div(int a, int b);

int main(int argc, char** argv){

    printf("Welcome to my basic, two number calculator.\n");

    int number1;
    char op[2];
    int number2;
    printf("Type in your first number: ");
    scanf("%d", &number1);
    printf("Type in your op: ");
    scanf("%s", op);
    printf("Type in your second number: ");
    scanf("%d", &number2);

    if(strcmp(op, "+") == 0){
        printf("%d\n",add(number1, number2));
    }
    else if(strcmp(op, "-")){
        printf("%d\n",sub(number1, number2));
    }
    else if( strcmp(op, "*")){
        printf("%d\n",mul(number1, number2));
    }
    else if( strcmp(op, "/")){
        printf("%d\n",div(number1, number2));
    }
    else{
        printf("That was not an operator we have!");
    }

    return 0;
}

int add(int a, int b) {
    return a+b;
}

int sub(int a, int b) {
    return a-b;
}

int mul(int a, int b) {
    return a*b;
}

int div(int a, int b) {
    return a/b;
}

File: Projects/03-Calculator/Rust/main.rs
use std::io;
use std::io::Write;

fn user_input_str() -> String {
    let mut buff = String::new();
    let _rl_result = io::stdin().read_line(&mut buff);
    buff.trim().to_string()
}

fn user_input_i32() -> i32 {
    let mut buff = String::new();
    let _rl_result = io::stdin().read_line(&mut buff);
    let to_ret : i32 = buff.trim().parse().unwrap();
    to_ret
}

fn main() {
    println!("Welcome to my basic, two number calculator.");
    
    print!("Type in your first number: ");
    io::stdout().flush().unwrap();
    let number1 = user_input_i32();

    print!("Type in your op: ");
    io::stdout().flush().unwrap();
    let operator = user_input_str();

    print!("Type in your second number: ");
    io::stdout().flush().unwrap();
    let number2 = user_input_i32();

    if operator == "+" {
        println!("{}", add(number1, number2));
    }
    else if operator == "-" {
        println!("{}", sub(number1, number2));
    }
    else if operator == "*" {
        println!("{}", mul(number1, number2));
    }
    else if operator == "/" {
        println!("{}", div(number1, number2));
    }
    else{
        println!("That was not an operator we have!");
    }

    
}

fn add(a : i32, b : i32) -> i32 {
    a+b
}
fn sub(a : i32, b : i32) -> i32 {
    a-b
}
fn mul(a : i32, b : i32) -> i32 {
    a*b
}
fn div(a : i32, b : i32) -> i32 {
    a/b
}